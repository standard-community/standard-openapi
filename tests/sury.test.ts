import * as S from "sury";
import { describe, expect, it } from "vitest";

import { toOpenAPISchema } from "~/index.js";

describe("sury", () => {
  it("basic", async () => {
    const schema = S.schema({
      myString: S.string,
      myUnion: S.union([S.number, S.boolean]),
    });

    const specs = await toOpenAPISchema(schema);
    expect(specs).toMatchSnapshot();
  });

  it("with metadata", async () => {
    const schema = S.extendJSONSchema(
      S.schema({
        myString: S.string,
        myUnion: S.union([S.number, S.boolean]),
      }).with(S.meta, {
        description: "My neat object schema",
      }),
      {
        $id: "MyNeatObjectSchema",
      },
    );

    const specs = await toOpenAPISchema(schema);
    expect(specs).toMatchSnapshot();
  });
});
