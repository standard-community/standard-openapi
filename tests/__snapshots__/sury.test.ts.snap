// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sury > basic 1`] = `
{
  "components": undefined,
  "schema": {
    "additionalProperties": true,
    "properties": {
      "myString": {
        "type": "string",
      },
      "myUnion": {
        "anyOf": [
          {
            "type": "number",
          },
          {
            "type": "boolean",
          },
        ],
      },
    },
    "required": [
      "myString",
      "myUnion",
    ],
    "type": "object",
  },
}
`;

exports[`sury > with metadata 1`] = `
{
  "components": {
    "schemas": {
      "MyNeatObjectSchema": {
        "additionalProperties": true,
        "description": "My neat object schema",
        "properties": {
          "myString": {
            "type": "string",
          },
          "myUnion": {
            "anyOf": [
              {
                "type": "number",
              },
              {
                "type": "boolean",
              },
            ],
          },
        },
        "required": [
          "myString",
          "myUnion",
        ],
        "type": "object",
      },
    },
  },
  "schema": {
    "$ref": "#/components/schemas/MyNeatObjectSchema",
  },
}
`;
